modules/lib/lib-event/src/main/resources/lib/xp/event.ts

```diff
@@ -0,0 +1,143 @@
+declare global {
+    interface XpLibraries {
+        '/lib/xp/event': typeof import('./event');
+    }
+}
+
+export interface ListenerParams<EventData extends object = EnonicEventData> {
+    /**
+     * Event type pattern
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * Callback event listener
+     */
+    callback: (event: EnonicEvent<EventData>) => void;
+
+    /**
+     * Local events only (default to false)
+     */
+    localOnly?: boolean;
+}
+
+export interface SendParams {
+    /**
+     * Event type
+     */
+    type: EnonicEventTypes;
+
+    /**
+     * `true` if it should be distributed in cluster
+     */
+    distributed?: boolean;
+
+    /**
+     * Additional data for event.
+     */
+    data?: object;
+}
+
+export interface EnonicEvent<EventData extends object = EnonicEventData> {
+    readonly type: EnonicEventTypes | string;
+    readonly timestamp: number;
+    readonly localOrigin: boolean;
+    readonly distributed: boolean;
+    readonly data: EventData;
+}
+
+export interface EnonicEventData {
+    readonly nodes: ReadonlyArray<EnonicEventDataNode>;
+    readonly state?: string; // event type <'node.stateUpdated'>
+}
+
+export interface EnonicEventDataNode {
+    readonly id: string;
+    readonly path: string;
+    readonly branch: string;
+    readonly repo: string;
+    readonly newPath?: string; // event type <'node.moved' | 'node.renamed'>
+}
+
+type LiteralUnion<T extends U, U = string> = T | (U & Record<never, never>);
+
+export type EnonicEventTypes =
```

Comment:
This name implies that Enonic XP only issues such events. Which is (soon) won't be true.
It affects all related types, too.
Either drop this handy thing, or rename the type(s). `EnonicNodeEventType`?

Reply:
Actually, it implies, that the `lib-event` can have any user-defined `string` event, but we also provide a list of [well-known events](https://developer.enonic.com/docs/xp/stable/api/lib-node#events) for autocompletion. I can see that we can also add `'node.manualOrderUpdated'`, that is not listed there.

Reply:
I would prefer EnonicNodeEventType | EnonicContentEventType | string
(EnonicContentEventType is not-existent just yet ,but its coming)

Reply:
Will replace with a `string`